### PR TITLE
Add support for distil-large-v3

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
The latest Distil-Whisper model, [distil-large-v3](https://huggingface.co/distil-whisper/distil-large-v3-ct2), is intrinsically designed to work with the OpenAI sequential algorithm.

This PR adds support for this checkpoint.

### CODE REVIEW

**Comment by BBC-Esq** [comment]
Nice to see some comradare.

**Comment by sanchit-gandhi** [comment]
cc @trungkienbkhn - this one's ready for review!

File: README.md [hunk 1]
```diff
@@ -159,18 +159,25 @@ for segment in segments:
 segments, _ = model.transcribe("audio.mp3")
 segments = list(segments)  # The transcription will actually run here.
 ```
-### Faster-distil-whisper
-For usage of `faster-distil-whisper`, please refer to: https://github.com/guillaumekln/faster-whisper/issues/533
+### Faster Distil-Whisper
+
+The Distil-Whisper checkpoints are compatible with the Faster-Whisper package. In particular, the latest [distil-large-v3](https://huggingface.co/distil-whisper/distil-large-v3)
+checkpoint is designed to give one-to-one equivalence with the Faster-Whisper transcription algorithm. The following code snippet
```
Comment (sanchit-gandhi) [inline README.md:165]:
```suggestion
checkpoint is intrinsically designed to work with the Faster-Whisper transcription algorithm. The following code snippet 
```

**Comment by sanchit-gandhi** [comment]
Hey @trungkienbkhn! The model is now live under https://huggingface.co/distil-whisper/distil-large-v3-ct2

Feel free to merge this PR at your convenience to enable Faster-Whisper support!

**Comment by trungkienbkhn** [comment]
@sanchit-gandhi , thanks for your contribution.
After testing with the example provided, I found that the transcription time for the [fw-distil-large-v3](https://huggingface.co/distil-whisper/distil-large-v3-ct2) model has been significantly reduced while maintaining almost the same quality compared with [fw-large-v3](https://huggingface.co/Systran/faster-whisper-large-v3) model.

However, when I added the option `word_timestamps=True`, I encountered an error:
```python3
segments, info = model.transcribe("audio.mp3", condition_on_previous_text=False, language="en", word_timestamps=True)
```
```
Processing segment at 00:00.000
[1]    7829 segmentation fault (core dumped)
```

This is caused by wrong `alignment_heads` field (a list of `[layer, head]`) in the config.json file of [fw-distil-large-v3](https://huggingface.co/distil-whisper/distil-large-v3-ct2) model.
Since the distill-whisper only has 2 decode layers, fw is unable to extract word-level timestamps from layer index > 2.
Screenshot from 2024-03-22 14-24-42

So we should use the default alignment_heads for fw-distil-large-v3 model, same as [fw-distil-large-v2](https://huggingface.co/distil-whisper/distil-large-v3-ct2/blob/main/config.json) model. For more details on this logic, you can refer to the implementation in [ctranslate2](https://github.com/OpenNMT/CTranslate2/blob/d963499e6ed496526e68fb149c908cb2da1f0255/python/ctranslate2/converters/transformers.py#L929).
Could you re-verify and update this field? Tks.

**Comment by Purfview** [comment]
@trungkienbkhn
Could you make usual conversion to float16 and place it at https://huggingface.co/Systran/faster-distil-whisper-large-v3

**Comment by BBC-Esq** [comment]
Oh yeah, almost forgot, wouldn't the be necessary because the faster-whisper library automatically points to the Systran repository unless another one is explicitly specified?

**Comment by Vaibhavs10** [comment]
> Could you make usual conversion to float16 and place it at https://huggingface.co/Systran/faster-distil-whisper-large-v3

Brilliant idea! That way all the faster-whisper checkpoints remain in one org.
We can remove the preliminary `ct2` checkpoint in `distil-whisper` org and point the README and documentation to `SYSTRAN`.

**Comment by sanchit-gandhi** [comment]
Alignment heads updated according to those in `distil-large-v2`: https://huggingface.co/distil-whisper/distil-large-v3-ct2/discussions/1

Note that we'd get a better alignment by inspecting the alignment from the DTW algorithm, e.g. as done [here](https://github.com/openai/whisper/blob/main/notebooks/Multilingual_ASR.ipynb).

However, the heuristic to use the last half layers for alignment by default should suffice for a first version.

**Comment by trungkienbkhn** [comment]
> Could you make usual conversion to float16 and place it at https://huggingface.co/Systran/faster-distil-whisper-large-v3

FYI, we have released a new ct2 conversion model (using float16) for distil-large-v3: https://huggingface.co/Systran/faster-distil-whisper-large-v3

File: faster_whisper/utils.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
     "distil-large-v2": "Systran/faster-distil-whisper-large-v2",
     "distil-medium.en": "Systran/faster-distil-whisper-medium.en",
     "distil-small.en": "Systran/faster-distil-whisper-small.en",
+    "distil-large-v3": "distil-whisper/distil-large-v3-ct2",
```
Comment (sanchit-gandhi) [inline faster_whisper/utils.py:28]: Swapping to the fp16 weights (cc @trungkienbkhn)

```suggestion
    "distil-large-v3": "Systran/faster-distil-whisper-large-v3",
```

**Comment by sanchit-gandhi** [comment]
Awesome, thanks @trungkienbkhn. I've updated this PR to use these fp16 weights.

**Comment by sanchit-gandhi** [comment]
Feel free to merge this PR at your convenience - it would be awesome to unblock faster-whisper for the distil-whisper community.

**Comment by cyaaronk** [comment]
Is distill-small supported?

**Pull request merged**